omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -257,6 +274,191 @@ else if(identifierSourceType.equals(Identifier_Pool)){
         }
         return ConversionUtil.convertToRepresentation(savedIdentifierSource, Representation.DEFAULT);         
     }
+    
+    @Override
+    public Object update(String uuid, SimpleObject updateBody, RequestContext context) throws ResponseException {
+        Object updatedIdentifierSource = null;
+        Object reservedIdentifiersToUpload = updateBody.get("reservedIdentifiers");
+        Object operation = updateBody.get("operation");
+        Object generateIdentifiers = updateBody.get("generateIdentifiers");
+        IdentifierSource identifierSourceToUpdate = Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(uuid);
+        SimpleObject output = new SimpleObject();
+        if (reservedIdentifiersToUpload != null) {
+            List<String> reservedIdentifiersList = new ArrayList<String>(Arrays.asList(reservedIdentifiersToUpload.toString().split(",")));
+            if(identifierSourceToUpdate != null){
+               for (int counter = 0; counter < reservedIdentifiersList.size(); counter++) {
+                    if (StringUtils.isNotBlank(reservedIdentifiersList.get(counter))) {
+                        identifierSourceToUpdate.addReservedIdentifier(reservedIdentifiersList.get(counter));
+                    }
+                }
+                Context.getService(IdentifierSourceService.class).saveIdentifierSource(identifierSourceToUpdate);
+                return output.add("reservedIdentifiers", identifierSourceToUpdate.getReservedIdentifiers());
```

Comment:
Why are you retuning these? Are they for any later use?

Reply:
Yes. I display the identifiers to the user.